assignments/prototypes.js

```diff
@@ -102,9 +140,112 @@
   console.log(archer.greet()); // Lilith offers a greeting in Elvish.
   console.log(mage.takeDamage()); // Bruce took damage.
   console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+
 
   // Stretch task: 
   // * Create Villian and Hero constructor functions that inherit from the Humanoid constructor function.  
   // * Give the Hero and Villians different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villian and one a hero and fight it out with methods!
\ No newline at end of file
+  // * Create two new objects, one a villian and one a hero and fight it out with methods!
+
+
+function Villian (vilAttrib) {
+  Humanoid.call(this, vilAttrib);
+  this.damage = vilAttrib.damage;
+}
+Villian.prototype = Object.create(Humanoid.prototype);
+
+Villian.prototype.weep = function(superhero) {
+  
+  if (superhero.hp > 0 && this.hp > 0) {
+    superhero.hp -= this.damage;
+    // console.log(superhero.takeDamage());
+    console.log (`${this.name} weeps on ${superhero.name} for ${this.damage} damage!!`);
+
+    if (superhero.hp <= 0) {
+      console.log(superhero.destroy());
+    }
+  }
+  else if (superhero.hp <= 0) {
+  }
+}
+
+
+// ================= Villian Above
+
+function Hero (heroAttrib) {
+  Humanoid.call(this, heroAttrib);
+  this.damage = heroAttrib.damage;
+}
+Hero.prototype = Object.create(Humanoid.prototype);
+
+Hero.prototype.slice = function(enemy) {
+  
+  if (enemy.hp > 0 && this.hp > 0) {   
+    enemy.hp -= this.damage; 
+    // console.log(enemy.takeDamage());
+    console.log (`${this.name} slices ${enemy.name} for ${this.damage} damage!!`);
+
+    if (enemy.hp <= 0) {
+      console.log(enemy.destroy());
+    }
+  }
+  else if (enemy.hp <= 0) {}
+}
+
+
+
+// ==================== Hero Above
+
+const gargamel = new Villian({
+  createdAt: new Date(),
+  dimensions: {
+    length: 1.5,
+    width: 2.5,
+    height: 5,
+  },
+  hp: 21,
+  name: "Gargamel",
+  faction: "Smurfville",
+  weapons: [
+    "Cat",
+  ],
+  language: "Smurf",
+  damage: 9,
+});
+
+
+const jack = new Hero({
+  createdAt: new Date(),
+  dimensions: {
+    length: 2.2,
+    width: 3.7,
+    height: 5.3,
+  },
+  hp: 25,
+  name: "Samurai Jack",
+  faction: "Scotsman",
+  weapons: [
+    "Katana",
+  ],
+  language: "English",
+  damage: 5,
+});
+
+
+jack.slice(gargamel);
```

Comment:
You're going to be sorry you used gargamel here once we get to the Smurfs projects.  You think I'm kidding.